src/peer_digest.cc

```diff
@@ -699,7 +649,7 @@ peerDigestFetchedEnough(DigestFetchState * fetch, char *buf, ssize_t size, const
     const char *reason = nullptr;  /* reason for completion */
     const char *no_bug = nullptr;  /* successful completion if set */
     const int pdcb_valid = cbdataReferenceValid(fetch->pd);
-    const int pcb_valid = cbdataReferenceValid(fetch->pd->peer);
+    const int pcb_valid = fetch->pd->peer.valid() ? 1 : 0;
```

Comment:
If we have to change this potentially invalid `fetch->pd` dereference, let's fix it:

```suggestion
    const int pcb_valid = pdcb_valid && fetch->pd->peer.valid();
```

I would keep the explicit `int` but not bother with explicit `1` and `0`, but that is your call.

There are more, similar dereference problems like this in the code below, but fixing them is outside this PR scope IMO.

Reply:
Done.